python/dex/interop/jax/apply.py

```diff
@@ -361,16 +361,22 @@ def dex_call_evaluate_linearized_transpose(cotangents, *args, func_atom):
 
   assert len(args) % 2 == 0
   num_primals = len(args) // 2
+  # TODO: Make it possible to get the signature without compiling the function
+  native = get_compiled(func_atom)
+  assert len(args) == len(native.explicit_argument_signature)
   module = func_atom.module.copy()
 
-  primals, tangents = args[:num_primals], args[num_primals:]
-
-  # Helper functions to build strings of primal and tangent inputs.
-  def arg_string(prefix, index_set):
-    return " ".join(f"{prefix}{i}" for i in index_set)
+  implicit_args = []
+  name_to_ty = {}
+  for binder in native.argument_signature:
+    if binder.implicit:
+      implicit_args.append("{" + binder.name + "}")
```

Comment:
perhaps check that `not name_to_ty`? Otherwise you will change the order of implicit and explicit args

Reply:
Thought about it, and added (i) a simple test case where it changes the order harmlessly, and (ii) a comment explaining why I think such reordering is safe in all cases that won't also break elsewhere in the interop.

Reply:
I agree that this is harmless right now, but this is only because of the current limitations of export. It's not a fundamental restriction though, and I can imagine having us lift it at some point. While I agree that disallowing that permutation might be too strong, the right thing to do here would be to do hoisting checks (intersect the free vars of type annotations of implicit binders with the explicit binders).

Reply:
Added hoisting checks.